test/rulesets/LinearAlgebra/symmetric.jl

```diff
@@ -18,22 +18,33 @@
         @testset "rrule" begin
             # on old versions of julia this combination doesn't infer but we don't care as
             # it infers fine on modern versions.
-            check_inferred = !(VERSION <= v"1.5" && T <: ComplexF64 && SymHerm <: Hermitian)
+            check_inferred = true#!(VERSION <= v"1.5" && T <: ComplexF64 && SymHerm <: Hermitian)
 
             x = randn(T, N, N)
             ∂x = randn(T, N, N)
             ΔΩ = randn(T, N, N)
             @testset "back(::$MT)" for MT in (Matrix, LowerTriangular, UpperTriangular)
+                _ΔΩ = MT(ΔΩ)
                 rrule_test(
-                    SymHerm, MT(ΔΩ), (x, ∂x), (uplo, nothing);
-                    check_inferred = check_inferred
+                    SymHerm, _ΔΩ, (x, ∂x), (uplo, nothing);
+                    check_inferred=false,
                 )
+                if check_inferred
+                    @inferred (function (SymHerm, x, ΔΩ, ::Val{uplo}) where {uplo}
+                        return rrule(SymHerm, x, uplo)[2](ΔΩ)
+                    end)(SymHerm, x, MT(ΔΩ), Val(uplo))
+                end
             end
             @testset "back(::Diagonal)" begin
                 rrule_test(
                     SymHerm, Diagonal(ΔΩ), (x, Diagonal(∂x)), (uplo, nothing);
-                    check_inferred = check_inferred
+                    check_inferred=false,
```

Comment:
why setting this to false everywhere?
No matter the input combination?

Reply:
This test is kind of flawed, because we rely on `uplo` to be constant propagated through `test_rrule`, so I added a separate `@inferred` test instead.

Reply:
comment to that effect?

Reply:
Ok, will add.